example/example1/example1.go

```diff
@@ -0,0 +1,290 @@
+	fmt.Println("Listing Stores")
+	stores, err := fgaClient.ListStores(ctx).Execute()
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Stores Count: %d\n", len(stores.Stores))
+
+	// GetStore
+	fmt.Println("Getting Current Store")
+	currentStore, err := fgaClient.GetStore(ctx).Execute()
+	if err != nil {
+		return err
+	}
+	fmt.Println("Current Store Name: %v\n" + currentStore.Name)
+
+	// ReadAuthorizationModels
+	fmt.Println("Reading Authorization Models")
+	models, err := fgaClient.ReadAuthorizationModels(ctx).Execute()
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Models Count: %d\n", len(models.AuthorizationModels))
+
+	// ReadLatestAuthorizationModel
+	latestAuthorizationModel, err := fgaClient.ReadLatestAuthorizationModel(ctx).Execute()
+	if err != nil {
+		return err
+	}
+	if latestAuthorizationModel.AuthorizationModel != nil {
+		fmt.Printf("Latest Authorization Model ID: %v\n", (*latestAuthorizationModel.AuthorizationModel).Id)
+	} else {
+		fmt.Println("Latest Authorization Model not found")
+	}
+
+	// WriteAuthorizationModel
+	fmt.Println("Writing an Authorization Model")
+	body := client.ClientWriteAuthorizationModelRequest{
+		SchemaVersion: "1.1",
+		TypeDefinitions: []openfga.TypeDefinition{
+			{
+				Type:      "user",
+				Relations: &map[string]openfga.Userset{},
+			},
+			{
+				Type: "document",
+				Relations: &map[string]openfga.Userset{
+					"writer": {This: &map[string]interface{}{}},
+					"viewer": {Union: &openfga.Usersets{
+						Child: []openfga.Userset{
+							{This: &map[string]interface{}{}},
+							{ComputedUserset: &openfga.ObjectRelation{
+								Object:   openfga.PtrString(""),
+								Relation: openfga.PtrString("writer"),
+							}},
+						},
+					}},
+				},
+				Metadata: &openfga.Metadata{
+					Relations: &map[string]openfga.RelationMetadata{
+						"writer": {
+							DirectlyRelatedUserTypes: &[]openfga.RelationReference{
+								{Type: "user"},
+								{Type: "user", Condition: openfga.PtrString("ViewCountLessThan200")},
+							},
+						},
+						"viewer": {
+							DirectlyRelatedUserTypes: &[]openfga.RelationReference{
+								{Type: "user"},
+							},
+						},
+					},
+				},
+			},
+		},
+		Conditions: &map[string]openfga.Condition{
+			"ViewCountLessThan200": {
+				Name:       "ViewCountLessThan200",
+				Expression: "ViewCount < 200",
+				Parameters: &map[string]openfga.ConditionParamTypeRef{
+					"ViewCount": {
+						TypeName: openfga.INT,
+					},
+					"Type": {
+						TypeName: openfga.STRING,
+					},
+					"Name": {
+						TypeName: openfga.STRING,
+					},
+				},
+			},
+		},
+	}
+	authorizationModel, err := fgaClient.WriteAuthorizationModel(ctx).Body(body).Execute()
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Authorization Model ID: %v\n", authorizationModel.AuthorizationModelId)
+
+	// ReadAuthorizationModels - after Write
+	fmt.Println("Reading Authorization Models")
+	models, err = fgaClient.ReadAuthorizationModels(ctx).Execute()
```

Comment:
Do we want to promote user to always limit number of models in read auth models?